### PR TITLE
added sink tokens to backward attention kernel

### PR DESCRIPTION
**Pull request by aws-ritikadm**
### Issue #, if available:

GPT-OSS uses sink tokens in attention. Hence we need to extend the flash attention backward kernel to support this feature.

### Description of changes:
In GPT-OSS, attention sinks are learned per-head bias logits that are appended to the attention scores. This PR adds functionality to compute the gradients for these tokens and unit tests for the same.
https://quip-amazon.com/GEvxAxikkl67/Attention-with-Sinks is the doc with more details on background and implementation.

### Testing:

Please see detailed unit test requirements in the [CONTRIBUTING.md](https://github.com/aws-neuron/nki-samples/blob/main/CONTRIBUTING.md)

- [x] The change is covered by numeric check using `nki.baremetal`
- [x] The change is covered by performance benchmark test using `nki.benchmark`
- [ ] The change is covered by end-to-end integration test

### Pull Request Checklist

- [x] I have filled in all the required field in the template
- [x] I have tested locally that all the tests pass
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the MIT-0 license.

### CODE REVIEW

File: src/nki_samples/reference/attention.py [hunk 1]
```diff
@@ -595,6 +597,8 @@ def flash_attn_bwd(
                           buffer=nl.shared_hbm)
   out_dv_ref = nl.ndarray((bs, nheads, d_head, seqlen_k), dtype=q_ref.dtype,
                           buffer=nl.shared_hbm)
+  out_dsinks_ref = nl.ndarray((bs, nheads), dtype=mixed_dtype, 
+                          buffer=nl.shared_hbm)
```
[the remark this answers is not in the document]
Comment (ankurkx-aws) [reply src/nki_samples/reference/attention.py]: it happens with direct allocation (although we are not using that in current kernel). see more details in first point under known limitations section [here](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/nki/nki_direct_allocation_guide.html#:~:text=All%20tensors%20declared%20with%20buffer%3Dnl.shared_hbm%20must%20be%20returned%20as%20the%20result%20of%20the%20kernel)

**Review by ankurkx-aws — APPROVED** [review]

**Review by tonyz0x0 — APPROVED** [review]
